data-prepper-plugins/otel-trace-raw-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/oteltrace/OTelTraceRawPrepper.java

```diff
@@ -6,59 +6,76 @@
 @DataPrepperPlugin(name = "otel_trace_raw_prepper", type = PluginType.PREPPER)
 public class OTelTraceRawPrepper extends AbstractPrepper<Record<ExportTraceServiceRequest>, Record<String>> {
 
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
-    private static final String INSTRUMENTATION_LIBRARY_SPANS = "instrumentationLibrarySpans";
-    private static final String INSTRUMENTATION_LIBRARY = "instrumentationLibrary";
-    private static final String SPANS = "spans";
-    private static final String RESOURCE = "resource";
-    private static final String ATTRIBUTES = "attributes";
-    private static final String START_TIME_UNIX_NANOS = "startTimeUnixNano";
-    private static final String END_TIME_UNIX_NANOS = "endTimeUnixNano";
-    private static final String START_TIME = "startTime";
-    private static final String END_TIME = "endTime";
-    private static final BigDecimal MILLIS_TO_NANOS = new BigDecimal(1_000_000);
-    private static final BigDecimal SEC_TO_MILLIS = new BigDecimal(1_000);
+    private static final long SEC_TO_MILLIS = 1_000L;
     private static final Logger log = LoggerFactory.getLogger(OTelTraceRawPrepper.class);
 
     public static final String SPAN_PROCESSING_ERRORS = "spanProcessingErrors";
     public static final String RESOURCE_SPANS_PROCESSING_ERRORS = "resourceSpansProcessingErrors";
     public static final String TOTAL_PROCESSING_ERRORS = "totalProcessingErrors";
 
+    private final long traceFlushInterval;
+    private final long rootSpanFlushDelay;
+
     private final Counter spanErrorsCounter;
     private final Counter resourceSpanErrorsCounter;
     private final Counter totalProcessingErrorsCounter;
 
+    // TODO: replace with file store, e.g. MapDB?
+    // TODO: introduce a gauge to monitor the size
+    private final Queue<DelayedParentSpan> delayedParentSpanQueue = new DelayQueue<>();
+    // TODO: replace with file store, e.g. MapDB?
+    // TODO: introduce a gauge to monitor the size
+    private final Map<String, RawSpanSet> traceIdRawSpanSetMap = new ConcurrentHashMap<>();
+
+    private long lastTraceFlushTime = 0L;
+
+    private final ReentrantLock gcLock = new ReentrantLock();
 
     //TODO: https://github.com/opendistro-for-elasticsearch/simple-ingest-transformation-utility-pipeline/issues/66
     public OTelTraceRawPrepper(final PluginSetting pluginSetting) {
         super(pluginSetting);
+        traceFlushInterval = SEC_TO_MILLIS * pluginSetting.getLongOrDefault(
+                OtelTraceRawPrepperConfig.TRACE_FLUSH_INTERVAL, OtelTraceRawPrepperConfig.DEFAULT_TG_FLUSH_INTERVAL);
+        rootSpanFlushDelay = SEC_TO_MILLIS * pluginSetting.getLongOrDefault(
+                OtelTraceRawPrepperConfig.ROOT_SPAN_FLUSH_DELAY, OtelTraceRawPrepperConfig.DEFAULT_ROOT_SPAN_FLUSH_DELAY);
+        Preconditions.checkArgument(rootSpanFlushDelay <= traceFlushInterval,
+                "rootSpanSetFlushDelay should not be greater than missingRootSpanSetFlushInterval.");
```

Comment:
nit - the variable names in the error message don't match the actual variable names